index.js

```diff
@@ -1,20 +1,236 @@
-const readline = require('readline');
-const readlineInterface = readline.createInterface(process.stdin, process.stdout);
+
+const readline = require("readline");
+const readlineInterface = readline.createInterface(
+  process.stdin,
+  process.stdout
+);
 
 function ask(questionText) {
   return new Promise((resolve, reject) => {
     readlineInterface.question(questionText, resolve);
   });
 }
 
+
+class Room {
+  //constructor that makes the room
+  constructor(roomName, description, items, lock ) {
+    (this.roomName = roomName),
+      (this.description = description),
+      (this.items = items);
+      (this.lock = lock || false)
+  }
+  // function that changes your room if you are able to 
+  move(room){
+    if(validRoomChange[Room].includes(room)){
+      currentRoom = room
+    } else {
+      console.log(`You cannot enter room yet! \n you might need somthing`)
+    }
+  }
+  // changeRoom(newRoom) {
+  //     if (validRoomChange[Room].includes(newRoom)) {
+  //       currentRoom = newRoom
+  //       console.log(describe());
+  //     }
+  // }
+  // function that describes your room
+  describe(){
+    console.log(`You are in the ${this.roomName} this room ${this. description}`)
+  }
+}
+
+class player {
+  constructor(currentRoom, inventory) {
+    (this.currentRoom = currentRoom || `street`),
+      (this.inventory = inventory || [] );
+  }
+  describePlayer(input){
+    if (input === `I`){
+      console.log(`you have ${this.inventory} on you, and are in the ${this.currentRoom}`)
+    }
+  }
+}
+
+class Item {
+  constructor(name, description, action, takable) {
+    (this.name = name),
+      (this.description = description),
+      (this.action = action || `nothing happens`),
+      (this.takable = takable || false);
+  }
+
+  //take function
+  take() {
+    if (this.takable) {
+      inventory.push(this.name);
```

Comment:
`inventory` isn't how you would access the inventory on your `playerInventory` object. As it's written, you would access this by using `playerInventory.inventory`. This is because `playerInventory` is an object whose key `inventory` has the value you want to manipulate.